modules/search-pipeline-common/src/main/java/org/opensearch/search/pipeline/common/README.md

```diff
@@ -0,0 +1,183 @@
+      PipelineContext pipelineContext
+  ) throws Exception {
+      String field = ConfigurationUtils.readStringProperty(TYPE, tag, config, "field");
+      boolean ignoreMissing = ConfigurationUtils.readBooleanProperty(TYPE, tag, config, "ignore_missing", false);
+      return new DeleteFieldResponseProcessor(tag, description, ignoreFailure, field, ignoreMissing);
+  }
+}
+```
+
+In this example, we provide specific configurations for which field should be deleted and whether the processor should ignore attempts to remove a non-existent field.  
+
+5. Add the newly added search processor to the `SearchPieplineCommonModulePlugin` getter for the corresponding processor type.
+
+For the `DeleteFieldResponseProcessor`, you would modify the response processor getter to have:
+
+```
+@Override
+public Map<String, Processor.Factory<SearchResponseProcessor>> getResponseProcessors(Parameters parameters) { 
+  return Map.of(
+    RenameFieldResponseProcessor.TYPE, 
+    new RenameFieldResponseProcessor.Factory(),
+    DeleteFieldResponseProcessor.TYPE,
+    new DeleteFieldResponseProcessor.Factory()
+  );
+}
+```
+
+6. After creating a search processor, the processor is ready to be tested in a search pipeline. 
+
+To test your new search processor, you can make use of the test [`SearchPipelineCommonYamlTestSuiteIT`](../../../../../../../yamlRestTest/java/org/opensearch/search/pipeline/common). 
+
+Following the format of the YAML files in [`rest-api-spec.test.search_pipeline`](../../../../../../../yamlRestTest/resources/rest-api-spec/test/search_pipeline), you should be able to create your own YAML test file to exercise your new processor. 
+
+7. Finally, the processor is ready to used in a cluster. 
+
+To use the new processor, make sure the cluster is reloaded and that the new processor is accessible.
+
+The new processor should show when calling `GET /_nodes/search_pipelines`.
+
+If the new processor is shown in the cURL response, the new processor should be available for use in a search pipeline.
+
+## Creating a Search Pipeline
+
+To create a search pipeline, you must create an ordered list of search processors in the OpenSearch cluster.
+
+An example creation request is:
+
+```
+PUT /_search/pipeline/my_pipeline 
+{
+  "request_processors": [
+    {
+      "filter_query" : {
+        "tag" : "tag1",
+        "description" : "This processor is going to restrict to publicly visible documents",
+        "query" : {
+          "term": {
+            "visibility": "public"
+          }
+        }
+      }
+    }
+  ],
+  "response_processors": [
+    {
+      "rename_field": {
+        "field": "message",
+        "target_field": "notification"
+      }
+    }
+  ]
+}
+```
+
+Alternatively, if you want to use just the `DeleteFieldResponseProcessor` created before, you would use:
+
+```
+PUT /_search/pipeline/my_pipeline2
+
+{
+  "response_processors": [
+    {
+      "delete_field": {
+        "field": "message" 
+      }
+    }
+  ]
+}
+```
```

Comment:
Should maybe follow that up with "To run a search request using the new processor, run `POST /myindex/_search?search_pipeline=my_pipeline2`"?

Reply:
Yeah sounds good. Let me add that.